projects/igniteui-angular/src/lib/buttonGroup/buttonGroup.component.ts

```diff
@@ -42,34 +42,180 @@ let NEXT_ID = 0;
 })
 
 export class IgxButtonGroupComponent implements AfterViewInit {
+    /**
+     * @hidden
+     */
     @ViewChildren(IgxButtonDirective) public buttons: QueryList<IgxButtonGroupComponent>;
 
+    /**
+     * An @Input property that sets the value of the `id` attribute.
+     * ```html
+     *  <igx-buttongroup [id]="'igx-dialog-56'" [multiSelection]="!multi" [values]="alignOptions">
+     * ```
+     */
     @HostBinding('attr.id')
     @Input()
     public id = `igx-buttongroup-${NEXT_ID++}`;
 
+    /**
+     * Allows you to set a style using the `itemContentCssClass` input.
+     *```typescript
+     *public style1 = "styleClass";
+     * //..
+     *```
+     * ```html
+     *<igx-buttongroup [itemContentCssClass]="style1" [multiSelection]="!multi" [values]="alignOptions">
+     *```
+     */
     @Input() set itemContentCssClass(value: string) {
         this._itemContentCssClass = value || this._itemContentCssClass;
     }
+
+    /**
+     * Returns the cssClass of the `igx-buttongroup`.
+     *```typescript
+     *@ViewChild("MyChild")
+     *public buttonG: IgxButtonGroupComponent;
+     *ngAfterViewInit(){
+     *   let buttonSelect = this.buttonG.itemContentCssClass;
+     *}
+     *```
+     */
     get itemContentCssClass(): string {
         return this._itemContentCssClass;
     }
+
+    /**
+     * An @Input property that enables selecting multiple buttons. By default, multiselection is false.
+     * ```html
+     * <igx-buttongroup [multiSelection]="false" [alignment]="alignment"></igx-buttongroup>
+     * ```
+     */
     @Input() public multiSelection = false;
+    /**
+     * An @Input property that allows setting the buttons in the button group.
+     * ```typescript
+     *  public ngOnInit() {
+     *      this.cities = [
+     *        new Button({
+     *          label: "Sofia"
+     *      }),
+     *        new Button({
+     *          label: "London"
+     *      }),
+     *        new Button({
+     *          label: "New York",
+     *          selected: true
+     *      }),
+     *        new Button({
+     *          label: "Tokyo"
+     *      })
+     *  ];
+     *  }
+     *  //..
+     * ```
+     * ```html
+     *  <igx-buttongroup [multiSelection]="false" [values]="cities"></igx-buttongroup>
+     * ```
+     */
     @Input() public values: any;
```

Comment:
@alexalex972 We should define a concrete return type. You should contact the person that wrote the code.